web-animations-2/Overview.bs

```diff
@@ -805,6 +849,62 @@ The procedure to [=pause an animation=] needs to refer not only to the
 <ins cite="#amend-CE">Likewise, the procedure to [=pause an animation=] needs to include scheduling
 a task for updating [=custom effects=].</ins>
 
+Update the constraints for setting the seek time to only apply when using a
+monotonic timeline. A [=scroll-driven animation=] needs to defer setting the
+hold time until the animation range has been computed.
+
+Replace:
+
+> If |seek time| is [=unresolved|resolved=],
+>
+>   <dl class=switch>
+>
+>       : If |has finite timeline| is true,
+>       ::
+>           Set |animation|'s [=animation/start time=] to |seek time|.
+>
+>       : Otherwise,
+>       ::
+>           Set |animation|'s [=hold time=] to |seek time|.
+>
+>   </dl>
+
+With:
+
+> If |seek time| is [=unresolved|resolved=],
+>
+>   <dl class=switch>
+>
+>       : If |has finite timeline| is true,
+>       ::
+>           Set the |auto align start time| flag to true.
```

Comment:
This seems odd. If auto align start time was false, wouldn't we want to continue using the modified start time? Doing this for non-aligned animations could result in a big jump to the time it pauses at even if there was no layout change.

I'm not sure I fully understand how non aligned start times work in this ecosystem - it seems like we should have a well defined way to preserve a non-aligned start time but also have simple ways to re-align it.

Reply:
Good catch, the condition for setting the flag should have required that start and hold time were both unresolved.  Refactored the section a bit to avoid nesting conditionals.  Basically if seek is set and the timeline is monotonic, we set the hold time to the seek time. If both hold and start time are unresolved, we must have a non-monotonic timeline and should set the auto align start time flag. 

Non-aligned start times are used after explicitly setting the start or current time. So if we set the current time on a play or pause pending animation, the current time will be preserved when the animation resolves its ready promise.
Playing an animation explicitly resets the "stickiness" as otherwise we have no way of doing so short of cancelling the animation. Pause pending animations only require an auto-aligned start time to compute the correct hold time. If we already have a time when pause-pending, we simply hold that time.

Reply:
I think it'd be reasonable to require canceling the animation to reset the start time stickiness. It seems odd that calling pause, then play could change the start time.

Reply:
Updated playing-an-animation to only set the |auto align start time| flag to true if |previous current time| is unresolved.
Updated setting-the-timeline to set |auto align start time| to true if |to finite timeline| is true and |from finite timeline| is false.  This also handles transitioning from a  null to a scroll timeline.